src/viewer.js

```diff
@@ -2919,8 +2885,29 @@ function onCanvasKeyPress( event ) {
     }
 }
 
+function onCanvasKeyPress( event ) {
+    var canvasKeyPressEventArgs = {
+      originalEvent: event.originalEvent,
+      preventDefaultAction: false,
+      preventVerticalPan: event.preventVerticalPan || !this.panVertical,
+      preventHorizontalPan: event.preventHorizontalPan || !this.panHorizontal
+    };
 
-
+    /**
+     * Raised when a keyboard key is pressed and the focus is on the {@link OpenSeadragon.Viewer#canvas} element.
+     *
+     * @event canvas-key-press
+     * @memberof OpenSeadragon.Viewer
+     * @type {object}
+     * @property {OpenSeadragon.Viewer} eventSource - A reference to the Viewer which raised this event.
+     * @property {Object} originalEvent - The original DOM event.
+     * @property {Boolean} preventDefaultAction - Set to true to prevent default keyboard behaviour. Default: false.
```

Comment:
```
@property {Boolean} preventVerticalPan - Set to true to prevent keyboard vertical panning. Default: false.
@property {Boolean} preventHorizontalPan - Set to true to prevent keyboard horizontal panning. Default: false.
```

This is okay
```
@property {Boolean} preventVerticalPan - They have no longer any effect.
@property {Boolean} preventHorizontalPan - They have no longer any effect.
```

Reply:
Yes, something like that  ;)  

 @iangilman what do you think? I'm drawing a blank on verbage lol

Reply:
@iangilman  I would actually be comfortable removing those two flags since it's all handled in onCanvasKeyDown now.

At the very least I would say remove the "Set to true to prevent keyboard xxx panning" text and replace with something like "No longer used, use the 'canvas-key' event instead"

🤷‍♂️

Reply:
Yeah, we should actually remove all three of the `prevent` properties and the documentation for them, since they don't do anything. This event is new since the last release, so there's no legacy API to consider. 

For that matter, perhaps we don't even need this event at all now that everything has been moved to keydown? I don't mind keeping it in, though.